tiles/src/main/java/com/protomaps/basemap/layers/PhysicalLine.java

```diff
@@ -18,30 +18,36 @@ public String name() {
   @Override
   public void processFeature(SourceFeature sf, FeatureCollector features) {
     if (sf.canBeLine() && (sf.hasTag("waterway") ||
-      sf.hasTag("natural", "strait", "cliff")) && (!sf.hasTag("waterway", "riverbank", "reservoir"))) {
+      sf.hasTag("natural", "strait", "cliff")) && (!sf.hasTag("waterway", "riverbank", "reservoir")))
+    {
+      int min_zoom = 12;
+      String kind = "other";
+      if (sf.hasTag("waterway")) {
+        kind = sf.getString("waterway");
+        if (sf.hasTag("waterway","river")) {
+          min_zoom = 9;
+        }
+      } else if (sf.hasTag("natural")) {
+        kind = sf.getString("natural");
+      }
+
       var feat = features.line(this.name())
         .setId(FeatureId.create(sf))
+        // Core Tilezen schema properties
+        .setAttr("pmap:kind", kind)
```

Comment:
Move towards Tilezen convention of single coalesce property across all OSM tags.